File: app/src/main/java/com/jakebarnby/filemanager/sources/onedrive/OneDriveFragment.java
package com.jakebarnby.filemanager.sources.onedrive;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jakebarnby.filemanager.R;
import com.jakebarnby.filemanager.sources.SourceFragment;
import com.jakebarnby.filemanager.glide.GlideApp;
import com.jakebarnby.filemanager.util.Constants;

/**
 * Created by Jake on 6/7/2017.
 */
public class OneDriveFragment extends SourceFragment {

    /**
     * Return a new instance of this Fragment
     *
     * @param sourceName The name of the source controlled by this fragment
     * @return A new instance of this fragment
     */
    public static SourceFragment newInstance(String sourceName) {
        SourceFragment fragment = new OneDriveFragment();
        Bundle args = new Bundle();
        args.putString("TITLE", sourceName);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mSource = new OneDriveSource(Constants.Sources.ONEDRIVE, this);
    }

    @Override
    public void onResume() {
        super.onResume();
        ((OneDriveSource)getSource()).checkForAccessToken(this);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        ((OneDriveSource)getSource()).getClient().handleInteractiveRequestRedirect(requestCode, resultCode, data);
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/onedrive/OneDriveFragment.kt
package com.jakebarnby.filemanager.sources.onedrive

import android.content.Intent
import android.os.Bundle
import androidx.core.os.bundleOf
import com.jakebarnby.filemanager.models.SourceType
import com.jakebarnby.filemanager.ui.sources.SourceFragment
import com.jakebarnby.filemanager.util.Constants

/**
 * Created by Jake on 6/7/2017.
 */
class OneDriveFragment : SourceFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        presenter.setFileSource(OneDriveSource(presenter))
    }

    override fun onResume() {
        super.onResume()
        (presenter.source as OneDriveSource).checkForAccessToken(this)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        (presenter.source as OneDriveSource).client?.handleInteractiveRequestRedirect(requestCode, resultCode, data)
    }

    companion object {
        fun newInstance(sourceId: Int): SourceFragment =
            OneDriveFragment().apply {
                arguments = bundleOf(
                    Constants.FRAGMENT_TITLE to SourceType.values()[sourceId].sourceName
                )
            }
    }
}